File: 38. Count and Say.cpp
#include <string>
class Solution {
public:
    string countAndSay(int n) {
        string res = "1";
        for (int i = 1; i < n; i++)
        {
            string temp = "";
            int c = 0;
            for (int j = 0; j < res.length(); j++)
            {
                if (j == 0)
                    c ++;
                else
                {
                    if (res[j] != res[j-1])
                    {
                        temp += to_string(c) + res[j-1];
                        c = 1;
                    }
                    else
                        c++;
                }
            }
            temp += to_string(c) + res[res.length()-1];
            res = temp;
        }
        return res;
    }
};

File: 38. Count and Say.py
class Solution(object):
    def countAndSay(self, n):
        """
        :type n: int
        :rtype: str
        """
        res = "1"
        for i in xrange(1,n):
            temp = ""
            count = 0
            for j in xrange(len(res)):
                if j == 0:
                    count +=1
                else:
                    if res[j] != res[j-1]:
                        temp += str(count) + res[j-1]
                        count = 1
                    else:
                        count += 1
            temp += str(count) + res[len(res)-1]
            res = temp
        return res
